### PR TITLE
Implement finding shortest path on undirected graph

### PR DESCRIPTION
**Pull request by Minh-Khang**
Fix issues: 
- Incorrect A* pathfinding with undirected graph #11
- get_shortest_path doesn't work correctly on undirected graphs #37

### CODE REVIEW

File: lib/graph/pathfinding.ex [hunk 1]
```diff
@@ -130,6 +167,54 @@ defmodule Graph.Pathfinding do
...
+            do_bfs(q2, g, target_id, tree, hfun)
+          end
+        end
+
+      {:empty, _} ->
+        nil
+    end
+  end
+
   defp construct_path(v_id_tree, %Graph{vertices: vs_tree, out_edges: oe_tree} = tree, path) do
```
Comment (achan1989) [inline lib/graph/pathfinding.ex:218]: Your undirected-specific version of `do_bfs` relies on `construct_path`, which only considers outgoing edges.
Compare this to your `all_edges` function, which looks at both incoming and outgoing edges.  Does the current implementation of `construct_path` do what you need it to do?  (I'm not saying you're wrong, I'm just struggling to understand how this all works).

Comment (Minh-Khang) [reply lib/graph/pathfinding.ex:218]: Hi. Thanks for asking. `construct_path` likes a backtracking tree to find out the path. You can see it has variable `tree` which is created on `a_star` function and it's not related to `g` as graph which uses in `all_edges` function. `tree` variable is created by adding the neighbors of a vertex which has min weight to the current `do_bfs` (`PriorityQueue` is a ascending queue of edges weight)

**Comment by Will-W** [comment]
Is there any chance of having this merged? This update is pretty important for what we're doing (I work with @achan1989).

Thanks!

File: lib/graph/pathfinding.ex (hunk 1 above)
Comment (achan1989) [reply lib/graph/pathfinding.ex:218]: Cool, and thanks for making this patch :)

**Comment by bitwalker** [comment]
Hey, thanks for the PR! Sorry I missed the updates, I'll get a new release out ASAP

**Pull request merged**